### PR TITLE
Added SignedMessage and ImmediateAuthorized Message and Service types

### PR DESCRIPTION
**Pull request by madhavajay**
## Description
- Added SignedMessage handling to Nodes and Clients
- Added HelloRoot service and messages for testing SignedMessage / Auth
- Updated Early Dev notebook with example of SignedMessage Auth
- Fixed build_proto script not clearing proto dir (see removed __init__)
- Changed Device Type / Spec to dataclass and added tests

## Affected Dependencies
None

## How has this been tested?
Local and CI

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by madhavajay** [comment]
My life before rebasing: 😂
Screen Shot 2020-08-12 at 7 09 23 pm

File: setup.cfg [hunk 1]
```diff
@@ -93,7 +93,7 @@ extras = True
 # in order to write a coverage file that can be read by Jenkins.
 # TODO (gmuraru) change cov-fail-under as soon as possible
 addopts =
-    --cov syft --cov-report term-missing --cov-fail-under 83 --no-cov-on-fail
```
Comment (iamtrask) [inline setup.cfg:96]: Any particular reason this was lowered?

Comment (iamtrask) [reply setup.cfg:96]: (I'm looking to push it upward as we add the codebase - in theory it shouldn't ever go in the other direct, yes?)

Comment (madhavajay) [reply setup.cfg:96]: Yeah sorry about this, I definitely tried by picking some untested code and doing a quick refactor + test and got it over 83% but then when I rebased, it went under 83% again and I needed to go have dinner. If codecov on one test fails the other different python 3.6/3.7/3.8 tests get abandoned early, so I figured it was better to bend the code cov slightly and make sure the code works than have my code merged hastily without the tests passing. I wonder if a solution might be to fail on codecov after all platform tests have been checked?

**Pull request merged**